### PR TITLE
Move 'app_manager/types_by_module' call to ES

### PR DESCRIPTION
**Pull request by esoergel**
I'm getting rid of the only remaining usage of 'app_manager/types_by_module' to delete the view (separately from this PR)
Commit-by-commit should make sense.
This requires an elasticsearch reindex
@orangejenny

### CODE REVIEW

File: corehq/pillows/mappings/app_mapping.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
 from corehq.util.elastic import es_index
 
-APP_INDEX = es_index("hqapps_12f505392bcec1a43d1c2c9ef9e0cb51")
-APP_MAPPING={'_meta': {'comment': 'Autogenerated [corehq.apps.app_manager.models.Application] mapping from ptop_generate_mapping 11/07/2013',
-           'created': None},
+APP_INDEX = es_index("hqapps_2016-01-05_1821")
+APP_MAPPING={'_meta': {'created': None},
```
Comment (dimagimon) [inline corehq/pillows/mappings/app_mapping.py]: E225 missing whitespace around operator

File: corehq/pillows/mappings/app_mapping.py [hunk 2]
```diff
@@ -56,7 +55,11 @@
                                     'type': 'date'},
                 'description': {'type': 'string'},
                 'doc_type': {'index': 'not_analyzed', 'type': 'string'},
-                'domain': {'type': 'string'},
+                'domain': {'fields': {'domain': {'index': 'analyzed',
+                                                 'type': 'string'},
+                                      'exact': {'index': 'not_analyzed',
+                                                'type': 'string'}},
+                           'type': 'multi_field'},
```
Comment (esoergel) [inline corehq/pillows/mappings/app_mapping.py]: This was really confusing to track down - it should have always been like this.

File: corehq/pillows/mappings/app_mapping.py [hunk 3]
```diff
@@ -70,7 +73,11 @@
                                                                                   'type': 'object'},
                                                                         'show': {'type': 'boolean'}},
                                                          'type': 'object'},
-                                           'case_type': {'type': 'string'},
+                                           'case_type': {'fields': {'case_type': {'index': 'analyzed',
+                                                                                  'type': 'string'},
+                                                                    'exact': {'index': 'not_analyzed',
+                                                                              'type': 'string'}},
+                                                         'type': 'multi_field'},
```
Comment (esoergel) [inline corehq/pillows/mappings/app_mapping.py]: I mentioned it in my commit message for this change, but basically I'm making a "not_analyzed" version of this field so the faceted results will be case sensitive, like it currently is with couch.

File: corehq/pillows/mappings/app_mapping.py (hunk 2 above)
Comment (esoergel) [inline corehq/pillows/mappings/app_mapping.py]: Both this change and the other will retain the same behavior for existing filters acting on `"domain"`, only `"domain.exact"` will use the non-analysed index.

**Comment by czue** [comment]
:tada:

**Pull request merged**